util/bitstream.go

```diff
@@ -226,3 +256,22 @@ func ParseUInt16(data []byte, bitStartIndex uint16) (uint16, error) {
 	}
 	return binary.BigEndian.Uint16([]byte{leftByte, rightByte}), nil
 }
+
+func (bs *BitStream) ReadTwoBitField(numFields int) ([]byte, error) {
+	result := []byte{}
```

Comment:
It would be better to make a byte array of the desired size. We know at the start how big this is going to be, and starting like this means `append()` will re reallocating memory as it expands the size of the array.

```
result := make([]byte, 0, numFields)
```
This will give you an empty slice with room to expand to `numFields` without triggering a new malloc when you append.

Reply:
Ah good point! I updated this to initialize with `numFields`.